internal/activity/pusher.go

```diff
@@ -0,0 +1,55 @@
+package activity
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"cdr.dev/coder-cli/internal/entclient"
+	"go.coder.com/flog"
+)
+
+const pushInterval = time.Minute
+
+// Pusher pushes activity metrics no more than once per pushInterval. Pushes
+// within the same interval are a no-op.
+type Pusher struct {
+	Source string
+	EnvID  string
+	Client *entclient.Client
+
+	state int64
```

Comment:
I think any time a CAS is used a good justification should be nearby. I see the code is trying to achieve a single flight, and so a mutex is avoided. A more readable alternative would be https://pkg.go.dev/golang.org/x/time/rate?tab=doc.